relayer/validate.go

```diff
@@ -0,0 +1,127 @@
+package relayer
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/athanorlabs/go-relayer/impls/gsnforwarder"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+
+	"github.com/athanorlabs/atomic-swap/coins"
+	"github.com/athanorlabs/atomic-swap/common/types"
+	contracts "github.com/athanorlabs/atomic-swap/ethereum"
+	"github.com/athanorlabs/atomic-swap/net/message"
+)
+
+func validateClaimRequest(
+	ctx context.Context,
+	request *message.RelayClaimRequest,
+	ec *ethclient.Client,
+	ourSFContractAddr ethcommon.Address,
+) error {
+	err := validateClaimValues(ctx, request, ec, ourSFContractAddr)
+	if err != nil {
+		return err
+	}
+
+	return validateClaimSignature(ctx, ec, request)
+}
+
+// validateClaimValues validates the non-signature aspects of the claim request:
+//  1. the claim request swap factory contract is byte compatible with ours
+//  2. the forwarder in the claim request swap factory contract has an identical
+//     address with our forwarder
+//  3. the relayer fee is equal to or greater than the passed minFee
```

Comment:
this isn't checked anymore

Reply:
I removed (3). The fee is no longer part of the claim request, and if we change the fee, we'll need to update `relayProtocolID`. If, for some reason, the 2 sides are not in agreement on the fee, the relayer won't send the transaction, as the signature will not validate.